src/agora/dobeamweights.cc

```diff
@@ -26,8 +26,8 @@ DoBeamWeights::DoBeamWeights(
     PtrGrid<kFrameWnd, kMaxDataSCs, complex_float>& ul_beam_matrices,
     PtrGrid<kFrameWnd, kMaxDataSCs, complex_float>& dl_beam_matrices,
     PhyStats* in_phy_stats, Stats* stats_manager,
-    std::shared_ptr<CsvLog::MatLogger> dl_csi_logger,
-    std::shared_ptr<CsvLog::MatLogger> dl_beam_logger)
+    std::array<std::shared_ptr<CsvLog::MatLogger>, CsvLog::kMatLogs>&
```

Comment:
Not sure this makes sense to me.  You are passing in a reference to an array of shared pointers.   This doesn't share the memory properly as the shared pointer is owned somewhere else.